### ISSUE TITLE
function-sample-aws-custom: Not working with current AWS Lambda

### ISSUE THREAD

**Issue by qwh2001** [spring-cloud/spring-cloud-function#512]
Version: 3.1.0.BUILD-SNAPSHOT

Today I built and deployed this sample lambda in AWS. When I tested it and it got timed out.  From its CloudWatch log, I found it is because it cannot resolve the destination and the request/response kept repeating until timed out:

[2020-05-03 21:09:57.916] - 12 FINE [reactor-http-nio-1] --- org.springframework.cloud.function.context.catalog.BeanFactoryAwareFunctionRegistry: Applying type conversion on output value
[2020-05-03 21:09:57.953] - 12 FINE [reactor-http-nio-1] --- org.springframework.cloud.function.adapter.aws.LambdaDestinationResolver: Lambda invoming value: GenericMessage [payload=byte[32], headers={contentType=application/json, id=6ea850ac-a524-1b97-765e-0c63298635bc, timestamp=1588540197953}]
[2020-05-03 21:09:57.953] - 12 FINE [reactor-http-nio-1] --- org.springframework.cloud.function.adapter.aws.LambdaDestinationResolver: Lambda destination resolved to: unknown
[2020-05-03 21:09:57.954] - 12 INFO [reactor-http-nio-1] --- org.springframework.cloud.function.web.source.SupplierExporter: Posting to: unknown

Then I did research into the Spring Cloud Function source code, I figured out the following solution with works for me:

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
@SpringBootApplication
public class LambdaApplication
		implements ApplicationContextInitializer<GenericApplicationContext> {

	private static Log logger = LogFactory.getLog(LambdaApplication.class);

	public Function<GenericMessage<String>, GenericMessage<String>> uppercase() {
		return value -> {
			logger.info("Processing: " + value);
			return new GenericMessage<String>(value.getPayload().toUpperCase(), value.getHeaders());
		};
	}

	public static void main(String[] args) {
		FunctionalSpringApplication.run(LambdaApplication.class, args);
	}

	@Override
	public void initialize(GenericApplicationContext context) {
		context.registerBean("uppercase", FunctionRegistration.class,
				() -> new FunctionRegistration<>(uppercase()).type(
						FunctionType.from(String.class).to(String.class).message()));
	}
}
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<

This code changes the uppercase function to use type GenericMessage<String>, so that the lambda destination resolver can get the "lambda-runtime-aws-request-id" from the message.

Thanks.

**Comment by olegz** [comment]
Yes, we're aware about the `org.json` [issue](https://github.com/spring-cloud/spring-cloud-function/issues/508) and I am looking at the `SupplierExporter: Posting to: unknown`. We actually saw similar issue lately and it was addressed. 
Also, would it be at all possible for you to push a reprodicible sample on GitHub.

**Comment by qwh2001** [comment]
I am using the sample project in your GitHub to reproduce this issue:

https://github.com/spring-cloud/spring-cloud-function/tree/3.0.x/spring-cloud-function-samples/function-sample-aws-custom

**Comment by danyflorez** [comment]
This error keeps happening on last version 3.1.0.BUILD-SNAPSHOT @olegz
